citadel/indico_citadel/backend.py

```diff
@@ -134,9 +133,12 @@ def upload_records(self, records, from_queue):
         )
         session.mount(self.search_app, HTTPAdapter(max_retries=retry))
         session.headers = self.headers
-        dumped_records = ((get_entry_type(rec), rec.id, self.dump_record(rec)) for rec in records)
+        dumped_records = (
+            (get_entry_type(rec), rec.id, self.dump_record(rec), records[rec] if from_queue else SimpleChange.created)
```

Comment:
I think `records[rec]` still doesn't make sense - records is a list/iterator and rec is an item from there...

Reply:
Also thought the same until I found out that while an initial export is a list/iterator, while doing a normal export, it returns a dict so I just kept it. We should probably unify both.